packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -211,7 +211,7 @@ export abstract class QueueProcessingServiceBase extends Construct {
     this.secrets = props.secrets;
 
     // Determine the desired task count (minimum) and maximum scaling capacity
-    this.desiredCount = props.desiredTaskCount || 1;
+    this.desiredCount = props.desiredTaskCount !== undefined ? props.desiredTaskCount : 1;
     this.maxCapacity = props.maxScalingCapacity || (2 * this.desiredCount);
```

Comment:
One thing you should handle as well is if the desiredTaskCount is 0 then maxScalingCapacity needs to be specified and it needs to be greater than 0

Reply:
Done